modules/devguide/examples/.clang-format

```diff
@@ -0,0 +1,11 @@
+# -*- mode: yaml; -*-
+
+BasedOnStyle: LLVM
+IndentWidth: 4
+BreakBeforeBraces: Linux
+ColumnLimit: 120
```

Comment:
This seems long,I thought C style guides recommend 80 mostly?

I see this is copied from `./modules/howtos/examples/.clang-format`, so fair enough if it reflects what we already have.

There's already a `./modules/devguide/examples/c/.clang-format` e.g. in `/c/` subdirectory, though it seems to be for cpp. Which one takes precedence?